rfd/0141-tsh-bench-db.md

```diff
@@ -0,0 +1,339 @@
+| `--uri`         | Direct database access URI. When provided, the benchmark will issue connections directly to this database, and no Teleport is involved in the testing. It must contain all the connection information, including authentication credentials. The contents from --db-user will be ignored when provided. |
+| `--connections` | The number of connections used to run the benchmark queries. If not provided, the value will be the same as `--rate`. |
+| `--create-sql`  | Specifies literal contents or a path to a SQL file to create database tables and load them with data. |
+| `--query-sql`   | Specifies literal contents or a path to a SQL file that contains query statements to benchmark against the database. |
+| `database`      | Teleport target database name. Available databases can be retrieved by running `tsh db ls`. If not provided, the `--uri` flag (for direct database connection) must be present. | 
+
+The `--rate` common flag will indicate how many queries per second will be
+established. For example, if you provide the flags `--connections=1 --rate=25`,
+the benchmark will send 25 queries per second using the same connection.
+
+Note: `--create-sql` and `--query-sql` are followed by `-sql` suffixes because
+receiving custom queries in plain text format will only be supported on SQL
+databases initially. Other database protocols, such as MongoDB, won’t support
+this format as their query DSL is not plain text.
+
+### Direct database connections
+To directly connect to the databases, the suites must accept the credentials
+that will be used. To avoid adding multiple flags to cover the scenarios, we're
+heading into having a single `--uri` flag, which will receive the database URI
+with all options included. This way, users can choose the authentication method
+that will be used.
+
+### TLS routing support
+The benchmark commands will support TLS routing Teleport deployments by running
+a local proxy (same flow as `tsh db connect`). Before the benchmark starts it
+will start the local proxy if necessary.
+
+```code
+$ tsh bench db connect --uri postgres://hello@localhost:5432
+```
+
+```mermaid
+sequenceDiagram
+  participant User
+  participant tsh
+  participant CP as Connection pool
+  participant Teleport
+  participant DB as Database
+
+  User->>tsh: tsh bench db query
+  tsh->>Teleport: Generate Certificates (tsh db login)
+
+  loop for N --connections
+    tsh->>Teleport: Establish connection
+    Teleport->>DB: Connect
+    DB->>tsh: Connection
+    tsh->>DB: Ping
+    tsh->>CP: Add a connection
+  end
+
+  Note over tsh,DB: Benchmark exeuction starts
+
+  loop every execution
+    tsh->>CP: Take on connection
+    tsh->>DB: Execute query
+  end
+```
+
+By providing this flag, the `tsh` will connect directly to the database without
+using any Teleport certificate.
+
```

Comment:
I think what's missing from the RFD is what do we actually measure? There are some examples of existing "tsh bench" functionality in the beginning but I don't see what actual metrics we'll be collecting for databases and how we'll be measuring performance.